src/cmd-build

```diff
@@ -207,41 +216,61 @@ fi
 
 # We want extraargs var to be split on words
 # shellcheck disable=SC2086
-/usr/lib/coreos-assembler/virt-install --dest="$(pwd)"/"${img_base}" \
-               --create-disk --kickstart-out "$(pwd)"/tmp/flattened.ks \
+/usr/lib/coreos-assembler/virt-install --dest="${PWD}"/"${img_base}" \
+               --create-disk --kickstart-out "${PWD}"/tmp/flattened.ks \
                --ostree-remote="${name}" --ostree-stateroot="${name}" \
                --ostree-ref="${ref:-${commit}}" \
                --location "${workdir}"/installer/*.iso \
                --ostree-repo="${workdir}"/repo ${extraargs-}
 
-/usr/lib/coreos-assembler/gf-anaconda-cleanup "$(pwd)"/"${img_base}"
-/usr/lib/coreos-assembler/gf-oemid "$(pwd)"/"${img_base}" "$(pwd)"/"${img_qemu}" qemu
+
+/usr/lib/coreos-assembler/gf-anaconda-cleanup "${PWD}"/"${img_base}"
+/usr/lib/coreos-assembler/gf-oemid "${PWD}"/"${img_base}" "${PWD}"/"${img_qemu}" qemu
 set +x
 # Clear the MCS SELinux labels
 # See https://github.com/coreos/coreos-assembler/issues/292
 chcon -vl s0 "${img_qemu}"
+
 # make a version-less symlink to have a stable path
 ln -s "${img_qemu}" "${name}"-qemu.qcow2
 
 img_qemu_sha256=$(sha256sum "${img_qemu}" | cut -f 1 -d ' ')
-
+img_qemu_size=$(stat --format=%s "${img_qemu}")
 build_timestamp=$(date -u +$RFC3339)
+vm_iso_checksum=$(awk '/SHA256.*iso/{print$NF}' "${workdir}"/installer/*CHECKSUM)
+
+src_location="container"
+if [ ! -f /lib/coreos-assembler/.clean ]; then
+    info "This version of coreos-assembler is running code from outside the container."
+    src_location="bind mount"
+fi
 
+# Record locations for code sources.
+# If the following condition is true, then /lib/coreos-assembler has been bind
+# mounted in and is using a different build tree.
+# shellcheck disable=SC2046 disable=SC2086
 cat > tmp/meta.json <<EOF
 {
  "buildid": "${buildid}",
  "name": "${name}",
  "summary": "${summary}",
  "coreos-assembler.build-timestamp": "${build_timestamp}",
- "coreos-assembler.image-input-checksum": "${image_input_checksum}",
- "coreos-assembler.image-genver": "${image_genver}",
  "coreos-assembler.image-config-checksum": "${image_config_checksum}",
+ "coreos-assembler.image-genver": "${image_genver}",
+ "coreos-assembler.image-input-checksum": "${image_input_checksum}",
+ "coreos-assembler.code-source": "${src_location}",
+ "coreos-assembler.vm-iso-checksum": "${vm_iso_checksum}",
+ "coreos-assembler.container-config-git": $(jq -M '.git' ${PWD}/coreos-assembler-config-git.json),
+ "coreos-assembler.image-git": $(jq -M '.git' /cosa/coreos-assembler-git.json),
```

Comment:
nit: `coreos-assembler.container-image-git` maybe? `image` is so overloaded these days.